### PR TITLE
{bio}[GCC/10.2.0] manta v1.6.0 w/ Python 3.8.6

### PR DESCRIPTION
**Pull request by deniskristak**
(created using `eb --new-pr`)

edit: requires patched Boost 1.74.0 from ~~#12007~~, to fix this issue:

```
/software/Boost/1.74.0-GCC-10.2.0/include/boost/serialization/set.hpp:40:33: error: library_version_type in namespace boost::serialization does not name a type; did you mean item_version_type?                                                                                                                                                                          
   40 |     const boost::serialization::library_version_type library_version(
      |                                 ^~~~~~~~~~~~~~~~~~~~
      |                                 item_version_type
/software/Boost/1.74.0-GCC-10.2.0/include/boost/serialization/set.hpp:47:30: error: library_version_type is not a member of boost::serialization; did you mean item_version_type?                                                                                                                                                                                         
   47 |     if(boost::serialization::library_version_type(3) < library_version){
      |                              ^~~~~~~~~~~~~~~~~~~~
      |                              item_version_type
/software/Boost/1.74.0-GCC-10.2.0/include/boost/serialization/set.hpp:47:56: error: library_version was not declared in this scope
   47 |     if(boost::serialization::library_version_type(3) < library_version){
      |                                                        ^~~~~~~~~~~~~~~
make[2]: *** [src/c++/lib/blt_util/CMakeFiles/manta_blt_util.dir/RegionTracker.cpp.o] Error 1
```

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3123.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/6e9c18f6ba42eec8052e2f12c0ccf8d3 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3501.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/9929e740d25ddefc81ca113aa2bb67b4 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3123.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/dda9473336e0bdf26af223aeff19ba98 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node2629.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/ecdf7d9581900dd2e199545f9173d4a5 for a full test report.

**Comment by boegel** [comment]
@deniskristak Test report failed on RHEL8 with:

```
*** Error compiling '/tmp/easybuild_build/manta/1.6.0/GCC-10.2.0/easybuild_obj/redist/pyflow-1.1.20/src/pyflow.py'...
  File "/tmp/easybuild_build/manta/1.6.0/GCC-10.2.0/easybuild_obj/redist/pyflow-1.1.20/src/pyflow.py", line 791
    dsfp = os.fdopen(os.open(taskDotScriptFile, os.O_WRONLY | os.O_CREAT, 0755), 'w')
                                                                             ^                                                                                                                              

SyntaxError: leading zeros in decimal integer literals are not permitted; use an 0o prefix for octal integers

make[2]: *** [redist/CMakeFiles/manta_pyflow.dir/build.make:83: redist/pyflow-1.1.20/src/pyflow.pyc] Error 1
```

**Comment by deniskristak** [comment]
@boegel can be tested again, fixed aforementioned error and added an extra piece for python3 compatibility

File: easybuild/easyconfigs/m/manta/manta-1.6.0-GCC-10.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    ('Doxygen', '1.8.20'),
+]
+
+dependencies = [
+    ('Python', '3.8.6'),
+    ('Boost', '1.74.0'),
+    ('zlib', '1.2.11'),
+]
+
+# preinstallopts = patch -i
```
Comment (boegel) [inline easybuild/easyconfigs/m/manta/manta-1.6.0-GCC-10.2.0.eb:34]: @deniskristak Please remove this commented out line?

File: easybuild/easyconfigs/m/manta/manta-1.6.0-GCC-10.2.0.eb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+dependencies = [
+    ('Python', '3.8.6'),
+    ('Boost', '1.74.0'),
+    ('zlib', '1.2.11'),
+]
+
+# preinstallopts = patch -i 
+# making included pyflow compatible with python3 as advised in https://github.com/Illumina/pyflow/issues/20
+preinstallopts = """sed -i "s/0755/0o755/g" ./redist/pyflow-1.1.20/src/pyflow.py && """
+preinstallopts += """sed -i "s/DataDirException, e/DataDirException(e)/g" ./redist/pyflow-1.1.20/src/pyflow.py && """
```
Comment (boegel) [inline easybuild/easyconfigs/m/manta/manta-1.6.0-GCC-10.2.0.eb:37]: @deniskristak Let's avoid using `"""` by using single quotes inside each command, which is safer anyway w.r.t. handling of special characters (there are none here, but fine)?

Also, maybe avoid hardcoding the version of the `pyflow` subdir by using `*`?

```python
preinstallopts = "sed -i 's/0755/0o755/g' ./redist/pyflow-*/src/pyflow.py && "
preinstallopts += "sed -i 's/DataDirException, e/DataDirException(e)/g' ./redist/pyflow-*/src/pyflow.py && "

File: easybuild/easyconfigs/m/manta/manta-1.6.0-GCC-10.2.0.eb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+"""
+
+toolchain = {'name': 'GCC', 'version': '10.2.0'}
+
+github_account = 'Illumina'
+source_urls = [GITHUB_LOWER_SOURCE]
+sources = ['v%(version)s.tar.gz']
+checksums = ['76ce7c4a95a3b92cb16381f9a3b091afa234948ae6f028982abbd49e21206421']
+
+# patches = ['manta-1.6_python3_compatibility.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/m/manta/manta-1.6.0-GCC-10.2.0.eb:22]: @deniskristak Please remove this commented out line?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/m/manta/manta-1.6.0-GCC-10.2.0.eb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        ['libexec/%s' % x for x in ['denovo_scoring.py', 'extractSmallIndelCandidates.py']],
+    ],
+    'dirs': ['libexec'],
+}
+
+modextrapaths = {
+    'PATH': 'libexec',
+    'PYTHONPATH': 'lib/python',
+}
+
```
Comment (smoors) [inline easybuild/easyconfigs/m/manta/manta-1.6.0-GCC-10.2.0.eb:53]: @deniskristak can you add sanity check commands for one or two of the binaries and python scripts?

**Review by smoors — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3501.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/5f11b330c383a934aa381108c0a4254d for a full test report.

**Comment by boegel** [comment]
@deniskristak The installation still fails for me, the problem already occurs during the `build` step (so using `sed` in`preinstallopts` doesn't help) is that `pyflow` is only unpacked *during* the build step.

One possible alternative way of fixing the issue could patching the `Makefile` in which the unpacking of `pyflow` is done?

**Comment by deniskristak** [comment]
@boegel closing this request, moved it to [https://github.com/easybuilders/easybuild-easyconfigs/pull/12359](url)

**Pull request closed without merging**